src/unix/freebsd.c

```diff
@@ -93,10 +94,18 @@ int uv_exepath(char* buffer, size_t* size) {
   mib[3] = -1;
 #endif
 
-  cb = *size;
-  if (sysctl(mib, 4, buffer, &cb, NULL, 0))
+  cb = sizeof abspath;;
+  if (sysctl(mib, 4, abspath, &cb, NULL, 0))
     return -errno;
-  *size = strlen(buffer);
+
+  cb -= 1;
```

Comment:
Renamed. I added an assert instead of a check, because if the sysctl succeeded that should always be > 0, right?